### PR TITLE
`hab sup help` no longer attempts to install 'core/hab-sup' if not present

### PR DESCRIPTION
**Pull request by jeremymv2**
Resolves https://github.com/habitat-sh/habitat/issues/5343

This PR introduces the following changes:

- Taking advantage of the one-way dependency of `hab-sup` on `hab`; we
 delcare the Supervisor clap commands in _one_ place, within the `hab`
 package. This keeps us DRY and allows for the next change below.
- All help commands for `hab sup` no longer will be routed to `hab-sup`
 but are handled by clap-rs within the `hab` package. This removes the odd behavior
 that has always existed where running any variation of `hab sup --help`
 would attempt to install `core/hab-sup` if not found; often requiring
 elevated or admin privileges.

The desired behavior can be tested manually:
1. remove any installation of `core/hab-sup`
2. compile then remove `./target/debug/hab-sup`
3. run `HAB_SUP_BINARY= ./target/debug/hab sup --help` or any help variation thereof under sup subcommand(s)
4. observe the help menu displayed without any of the previous messages like
```
∵ Missing package for core/hab-sup/0.65.0
» Installing core/hab-sup/0.65.0
☁ Determining latest version of core/hab-sup/0.65.0 in the 'stable' channel
...
```
Are there any ideas how we can build a test for this? I would be happy to write some tests. 

Signed-off-by: Jeremy J. Miller <jm@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by baumanj** [comment]
> Are there any ideas how we can build a test for this? I would be happy to write some tests.

I think you could do something similar to [the tests I've done here](https://github.com/habitat-sh/habitat/tree/master/test/end-to-end). Using `TESTING_FS_ROOT` with a temporary directory will give you a pristine environment. Compare:

```
➤ env RUST_LOG= TESTING_FS_ROOT=(mktemp -d) hab sup --help
DEBUG: setting custom filesystem root for testing only (TESTING_FS_ROOT='/tmp/tmp.9UsN1XmHqm')
∵ Missing package for core/hab-sup/0.65.0
» Installing core/hab-sup/0.65.0
☁ Determining latest version of core/hab-sup/0.65.0 in the 'stable' channel
☛ Verifying core/hab-sup/0.65.0/20181004221503
[…]
✓ Installed core/busybox-static/1.24.2/20170513215502
[…]
★ Install of core/hab-sup/0.65.0/20181004221503 complete with 13 new packages installed.
hab-sup 0.65.0/20181004221503

Authors: The Habitat Maintainers <humans@habitat.sh>
The Habitat Supervisor

USAGE:
[rest of help…]
```
with
```
env RUST_LOG= TESTING_FS_ROOT=(mktemp -d) hab sup --help ran for 8277 ms
➤ env RUST_LOG= TESTING_FS_ROOT=(mktemp -d) ./target/debug/hab sup --help
DEBUG: setting custom filesystem root for testing only (TESTING_FS_ROOT='/tmp/tmp.7YflQTBGqK')
hab-sup 0.68.0

Authors: The Habitat Maintainers <humans@habitat.sh>
The Habitat Supervisor

USAGE:
[rest of help…]
```

**Comment by jeremymv2** [comment]
Oh nice! I will add an integration test under `test/end-to-end/` in this PR. Thank you for point that out @baumanj 🥇

**Comment by jeremymv2** [comment]
I rebased the commits onto master following the 0.68.0 release. It looks like Appveyor timed out at 1hr though.

File: test/end-to-end/test_hab_help_doesnt_install_hab_sup.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# A simple test assertion that running `hab sup --help` will not
+# attempt to install `core/hab-sup` if that pkg is not present.
+
+set -ou pipefail
+IFS=$'\n\t'
+
```
[the remark this answers is not in the document]
Comment (baumanj) [reply test/end-to-end/test_hab_help_doesnt_install_hab_sup.sh]: We don't have automation for the end-to-end tests yet. We're planning to do that when we add test automation to Buildkite, but in the meantime it's good to be building up a library and they're useful for local development too.

**Comment by jeremymv2** [comment]
Some issues  (apparently unrelated to this change) started building up with SpecWatcher in the Appveyor tests here: https://ci.appveyor.com/project/chef/habitat/builds/20302407/job/bt45r5ida2kt9w4d#L1534

```
(SW): SpecWatcher(C:\Users\appveyor\AppData\Local\Temp\1\.tmpfQNKxK) could not start fs watching, ending thread (Input watch path is neither a file nor a directory.)
habitat_sup-a93c108b1c80df3d(SW): SpecWatcher(C:\Users\appveyor\AppData\Local\Temp\1\.tmpOcZ8aE) could not start fs watching, ending thread (Input watch path is neither a file nor a directory.)
```

**Comment by baumanj** [comment]
I'm confident enough those AppVeyor failures are unrelated that I'll go ahead and merge this. Thanks again for the great work!

**Pull request merged**